go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -342,17 +342,38 @@ func (vs *vstreamer) parseEvent(ev mysql.BinlogEvent) ([]*binlogdatapb.VEvent, e
 			vs.plans[id] = nil
 			return nil, nil
 		}
+
+		tableName := tm.Name
+		var cols []schema.TableColumn
+		for i, typ := range tm.Types {
+			t, err := sqltypes.MySQLToType(int64(typ), int64(tm.Flags))
+			if err != nil {
+				return nil, fmt.Errorf("unsupported type: %d, position: %d", typ, i)
+			}
+			cols = append(cols, schema.TableColumn{
+				Name: sqlparser.NewColIdent(fmt.Sprintf("@%d", i+1)),
+				Type: t,
+			})
+		}
 		st := vs.se.GetTable(sqlparser.NewTableIdent(tm.Name))
 		if st == nil {
-			return nil, fmt.Errorf("unknown table %v in schema", tm.Name)
-		}
-		if len(st.Columns) < len(tm.Types) {
-			return nil, fmt.Errorf("cannot determine table columns for %s: event has %d columns, current schema has %d: %#v", tm.Name, len(tm.Types), len(st.Columns), ev)
+			if !vs.filter.BestEffortNameInFieldEvent {
+				return nil, fmt.Errorf("unknown table %v in schema", tm.Name)
+			}
+		} else {
+			// preserving old behavior.
+			if len(st.Columns) < len(tm.Types) && !vs.filter.BestEffortNameInFieldEvent {
+				return nil, fmt.Errorf("cannot determine table columns for %s: event has %d columns, current schema has %d: %#v", tm.Name, len(tm.Types), len(st.Columns), ev)
+			}
+			tableName = st.Name.String()
+			if len(tm.Types) == len(st.Columns) || !vs.filter.BestEffortNameInFieldEvent {
```

Comment:
I tried to compare the types. Strict type check doesn't work. It seems from VARBINARY column, tm returns VARCHAR. Do we have type equivalence function that i use?